packages/core/contracts/financial-templates/perpetual-multiparty/PerpetualPositionManager.sol

```diff
@@ -33,6 +33,24 @@ contract PerpetualPositionManager is FundingRatePayer, FundingRateApplier {
      *  PRICELESS POSITION DATA STRUCTURES  *
      ****************************************/
 
+    // Define the contract's constructor parameters as a struct to enable more variables to be specified.
+    // This is required to enable more params, over and above Solidity's limits.
+    struct PMConstructorParams {
```

Comment:
I've had to do this because `coverage` adds constructor params (I think) and throws a Stack too deep error on compilation. But `test` works fine

Reply:
I ended up removing the requirement to add funding-rate-record-params to the Perp constructor params, but left this change here because we might want it in future if we add more constructor params